packages/sdk/src/sdk/api/remix-contests/RemixContestApi.ts

```diff
@@ -0,0 +1,90 @@
+import snakecaseKeys from 'snakecase-keys'
+
+import { LoggerService } from '../../services'
+import type { StorageService } from '../../services'
+import {
+  EntityManagerService,
+  EntityType,
+  Action
+} from '../../services/EntityManager/types'
+import { decodeHashId } from '../../utils/hashId'
+import {
+  Configuration,
+  RemixContestsApi as GeneratedRemixContestsApi
+} from '../generated/default'
+
+type RemixContestMetadata = {
+  eventId: number
+  trackId: number
+  endDate: string // ISO format date string
+}
+
+type ExtendedConfiguration = Configuration & {
+  userId?: number
+}
+
+export class RemixContestApi extends GeneratedRemixContestsApi {
+  constructor(
+    configuration: ExtendedConfiguration,
+    private readonly entityManager: EntityManagerService,
+    private readonly logger: LoggerService
+  ) {
+    super(configuration)
+    this.logger = logger.createPrefixedLogger('[playlists-api]')
+  }
+
+  async generateEventId() {
+    const response = await this.getUnclaimedEventID()
+    const { data: unclaimedId } = response
+    if (!unclaimedId) {
+      return Math.floor(Math.random() * 1000000)
+    }
+    return decodeHashId(unclaimedId)!
+  }
+
+  async createRemixContest(metadata: RemixContestMetadata) {
```

Comment:
Do you mind updating these to use the same zod/param validation pattern as the other POST apis? `PlaylistsApi` and `UsersApi` both have examples for how to set up the schemas and do the parsing (including the userId).